### ISSUE TITLE
Fix clippy lints in SSZ

### ISSUE THREAD

**Issue by paulhauner** [sigp/lighthouse#52]
## Description

The [clippy](https://github.com/rust-lang-nursery/rust-clippy) linter has a lot to say about our [ssz](https://github.com/sigp/lighthouse/tree/master/beacon_chain/utils/ssz) implementation. We'd like to remove these warnings so we can eventually start failing builds if they trigger clippy.

## Present Behaviour

Clippy outputs warnings when run against the ssz crate.

## Expected Behaviour

Clippy should not have any warnings.

## Steps to resolve

1. Run clippy
2. Fix lints

## Bonus points

- Tidy up the code (there's definitely some cases where things could be neater)
- Optimise the code

The crate was written during an experimental phase of SSZ and is not as elegant as we would like. We're not attached to it and would be open to suggestions on a restructure. Please make a new issue if you'd like to restructure :)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 180.0 DAI (180.0 USD @ $1.0/DAI)  attached to it as part of the Ethereum Community Fund via ECF Web 3.0 Infrastructure Fund fund.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/sigp/lighthouse/52/1547).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/sigp/lighthouse/52/1547).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $48,729.97 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 12 months from now.
Please review their action plans below:


**1) [mahim23](https://gitcoin.co/profile/mahim23) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/sigp/lighthouse/52/1547?mutate_worker_action=approve&worker=mahim23) | [reject worker](https://gitcoin.co/issue/sigp/lighthouse/52/1547?mutate_worker_action=reject&worker=mahim23))_.**

Hey, I want to start contributing to the project and this issue looks like a good place to start.
I have started looking at the codebase.
The plan is to run clippy::all against the codebase and ensure each warning is fixed.
Also I will try to look for optimizations in the encode and decode sub modules and implement them.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/sigp/lighthouse/52/1547).

**Comment by g-r-a-n-t** [comment]
I've got the lint errors popping up locally. Are there any tests around the existing functionality?

**Comment by mahim23** [comment]
Hey guys, I've started fixing the warnings. Can send a PR by tonight!

**Comment by g-r-a-n-t** [comment]
I've thinned out most of the warnings and added ignores for a couple tricky ones: https://github.com/sigp/lighthouse/compare/master...g-r-a-n-t:clippy-warnings

**Comment by mkosowsk** [comment]
@mahim23 and @g-r-a-n-t, great work! But in the future, please do not start Gitcoin bounties until you have been approved for the work. This way we can avoid two people doing the same work :)

@paulhauner is @g-r-a-n-t's PR what you had in mind?

@mahim23 how much work have you done on this bounty? Thanks!

**Comment by mahim23** [comment]
Hey @mkosowsk, I had just set up the repository and fixed a couple of warnings. Was waiting for approval.

**Comment by mahim23** [comment]
Hey @mkosowsk, I had just set up the repository and fixed a couple of warnings. Was waiting for approval.

**Comment by g-r-a-n-t** [comment]
@mkosowsk I'll avoid that in the future. Didn't mean to put you in a awkward position, just had some trouble stopping after I got a taste of Rust.

**Comment by mkosowsk** [comment]
I have approved @mahim23 as he was first to apply 👍

**Comment by mkosowsk** [comment]
@g-r-a-n-t you have been approved to start work

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done 

<hr>

__Work for 180.0 DAI (180.0 USD @ $1.0/DAI) has been submitted by__:

 1. [@g-r-a-n-t](https://gitcoin.co/profile/g-r-a-n-t)



@mkosowsk please take a look at the submitted work:
* [PR](https://github.com/sigp/lighthouse/pull/58) by @g-r-a-n-t

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/sigp/lighthouse/52/1547)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/sigp/lighthouse/52/1547).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $50,988.44 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by g-r-a-n-t** [comment]
@mkosowsk Could you please pay this out? The work has been merged.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__The funding of 180.0 DAI (180.0 USD @ $1.0/DAI)  attached to this issue has been approved & issued to @g-r-a-n-t.__  

 * Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/sigp/lighthouse/52/1547)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $57,009.49 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by paulhauner** [comment]
Thank you @g-r-a-n-t and @gitcoinbot :)
